msgraph/chat_test.go

```diff
@@ -0,0 +1,116 @@
+package msgraph_test
+
+import (
+	"fmt"
+	"github.com/manicminer/hamilton/internal/utils"
+	"testing"
+
+	"github.com/manicminer/hamilton/internal/test"
+	"github.com/manicminer/hamilton/msgraph"
+	"github.com/manicminer/hamilton/odata"
+)
+
+func TestChatClient(t *testing.T) {
+	c := test.NewTest(t)
+	defer c.CancelFunc()
+
+	self := testDirectoryObjectsClient_Get(t, c, c.Claims.ObjectId)
+
+	// To create a chat two users need to be assigned to the chat.
+	// An owner needs to be assigned
+	user1 := testUsersClient_Create(t, c, msgraph.User{
+		AccountEnabled:    utils.BoolPtr(true),
+		DisplayName:       utils.StringPtr("test-user1"),
+		MailNickname:      utils.StringPtr(fmt.Sprintf("test-user-%s", c.RandomString)),
+		UserPrincipalName: utils.StringPtr(fmt.Sprintf("test-user-%s@%s", c.RandomString, c.Connections["default"].DomainName)),
+		PasswordProfile: &msgraph.UserPasswordProfile{
+			Password: utils.StringPtr(fmt.Sprintf("IrPa55w0rd%s", c.RandomString)),
+		},
+	})
+	user2 := testUsersClient_Create(t, c, msgraph.User{
+		AccountEnabled:    utils.BoolPtr(true),
+		DisplayName:       utils.StringPtr("test-user2"),
+		MailNickname:      utils.StringPtr(fmt.Sprintf("test-user-%s", c.RandomString)),
+		UserPrincipalName: utils.StringPtr(fmt.Sprintf("test-user-%s@%s", c.RandomString, c.Connections["default"].DomainName)),
+		PasswordProfile: &msgraph.UserPasswordProfile{
+			Password: utils.StringPtr(fmt.Sprintf("IrPa55w0rd%s", c.RandomString)),
+		},
+	})
+
+	// Check that a group chat and a OneOnOne chat can be created
+	newChat := msgraph.Chat{
+		Topic:    utils.StringPtr(fmt.Sprintf("test-chat-%s", c.RandomString)),
+		ChatType: utils.StringPtr(msgraph.ChatTypeGroup),
+		Members: &[]msgraph.ConversationMember{
+			{
+				ID:    user1.Id,
```

Comment:
Since these are bind records, it looks like the odata type is required here, and also the ID needs to be sent as a full odata ID.

Reply:
I apologize for taking this PR on without understanding this part. 

Can you link an example of where the OdataID is passed in a full context? I must be missing the apparent examples elsewhere in the code. 

My understanding from your first part is that it needs to be: 
```
			{
                                 ODataType: odata.TypeConversationMember
				ID:    user1.Id,
				Roles: &[]string{"owner"},
			},
```
And then the odata.TypeConversationMember be added as a type?